File: c/cz/mg/vulkan/PFN_vkCreateDescriptorSetLayout.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_vk_PFNvkCreateDescriptorSetLayout_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pSetLayout, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkCreateDescriptorSetLayout f = (PFN_vkCreateDescriptorSetLayout)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkDescriptorSetLayoutCreateInfo*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkDescriptorSetLayout*)jniLongToPointer(pSetLayout))
    );
}

File: cz/mg/vulkan/PFNvkCreateDescriptorSetLayout.java
package cz.mg.vulkan;

public class PFNvkCreateDescriptorSetLayout extends VkFunctionPointer {
    public PFNvkCreateDescriptorSetLayout() {
    }

    public PFNvkCreateDescriptorSetLayout(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkCreateDescriptorSetLayout(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkCreateDescriptorSetLayout(long value) {
        setValue(value);
    }

    public PFNvkCreateDescriptorSetLayout(VkInstance instance) {
        super(instance, new VkString("vkCreateDescriptorSetLayout"));
    }

    public void call(VkDevice device, VkDescriptorSetLayoutCreateInfo pCreateInfo, VkAllocationCallbacks pAllocator, VkDescriptorSetLayout pSetLayout, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.NULL_ADDRESS, pCreateInfo != null ? pCreateInfo.getVkAddress() : VkPointer.NULL, pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL, pSetLayout != null ? pSetLayout.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.NULL_ADDRESS);
    }

    private static native void call(long vkaddress, long device, long pCreateInfo, long pAllocator, long pSetLayout, long rval);
}
